File: Quinta/radice_quadrata/Erone/radice1_err.cpp
#include <iostream>
#include <math.h>
#include <iomanip>

using namespace std;

#define EPS 0.001

double radice_q(double a, double x1){
    if((fabs(x1 - a/x1)/2) >= EPS){
        x1 = (x1 + a/x1)/2;
        return radice_q(a, x1);
    }
    else return x1;
}

int main(){
    double numero;
    cout.precision(15);
    cout << "Calcolo approssimato radice quadrata\n";
    cout << "-inserisci un valore positivo: ";
    cin >> numero;

    cout << "\nApprossimato: " << radice_q(numero, numero/2);
    cout << "\nPreciso     : " << sqrt(numero);
}

File: Quinta/radice_quadrata/Erone/Radice1Err.java
package radice_quadrata.Erone;
import java.util.Scanner;
public class Radice1Err {
    public static double EPS = 0.001;

    static double radice_q(double a, double x1){
        if((Math.abs(x1 - a/x1)/2) >= EPS)
        {
            x1 = (x1 + a/x1) / 2;
            return radice_q(a, x1);
        }
        else return x1;
    }

public static void main(String []arg){
    Scanner in = new Scanner(System.in);
    double numero;
    System.out.println("Calcolo approssimato radice \n");
    System.out.print("inserisci un valore positivo: ");
    numero = in.nextInt();
    System.out.println("Approssimato:" + radice_q(numero, numero/2));
    System.out.println("Preciso     :" + Math.sqrt(numero));
}
}